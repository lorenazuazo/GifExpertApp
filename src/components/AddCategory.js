import React,{ useState } from 'react';
import PropTypes from 'prop-types';


const AddCategory = ({setCategorias}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=> {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e)=> {
        e.preventDefault();

        //se paso como referencia la funcion setCategorias
        console.log('se llamo handelSubmit',inputValue)
        setCategorias(cat => [inputValue,...cat])
        setInputValue('');
    }

    return (
        <form onSubmit={handleSubmit}>
         <p>{inputValue}</p> {/*esto es solo para el test */}
            <input 
                type='text'
                value = {inputValue}
                onChange ={handleInputChange}
            />
        </form>
    )
}
AddCategory.propTypes = {
    setCategorias: PropTypes.func.isRequired

}
export default AddCategory;
