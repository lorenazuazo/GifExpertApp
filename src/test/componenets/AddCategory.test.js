import React from 'react';
import '@testing-library/jest-dom'//se importa para usar toHaveBeenCalled si no reconoce al escribir
import { shallow } from 'enzyme';
import  AddCategory  from  '../../components/AddCategory';

describe('Test al componente <AddCategory />', () => {

    const setCategorias =  jest.fn();
    let wrapper = shallow(<AddCategory setCategorias={setCategorias}/>);

    beforeEach(() =>{
        jest.clearAllMocks(); //esto limpia todas las simulaciones cada vez que entra en un test
        wrapper = shallow(<AddCategory setCategorias={setCategorias}/>);

    })
    test('El componente se debe mostrar correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de cambiar la caja de texto', () => {
        const input=wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change',{ target:{value} });

        expect(wrapper.find('p').text().trim()).toBe(value);
    });
    
    test('No debe de postear la informacion onSubmit', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategorias).not.toHaveBeenCalled();
    });

    test('debe de llamar a setCategorias y limpiar la caja de text', () => {
        const value = 'Lore';
        //1.simular wl inputChange
        wrapper.find('input').simulate('change',{target:{value}});
        //2.simular el submit
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        //3.setCategorias de haber sido llmado
        expect(setCategorias).toHaveBeenCalled();
        //tambien se puede hacer preguntando cuantas veces se llamo
        expect(setCategorias).toHaveBeenCalledTimes(1);
        //tambien puedo preguntar si fue llamado son una funcion
        expect(setCategorias).toHaveBeenCalledWith(expect.any(Function))
        const input = wrapper.find('input').prop('value');
       //4.el valor del input debe estar vacio
        expect(input).toBe('');


    });
    
    
    
})
