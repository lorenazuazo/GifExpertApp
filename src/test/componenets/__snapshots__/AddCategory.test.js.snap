// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test al componente <AddCategory /> El componente se debe mostrar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
   
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
